plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/Bom.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.libraries;
+
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.google.cloud.tools.libraries.json.CloudLibraryClient;
+import com.google.cloud.tools.libraries.json.CloudLibraryClientMavenCoordinates;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableSet.Builder;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import org.w3c.dom.Element;
+
+/**
+ * 
+ */
+class Bom {
+  
+  private final static XPathFactory factory = XPathFactory.newInstance();
+
+  private static Set<String> artifacts; 
+  
+  static {
+    try {
+      List<CloudLibrary> cloudLibraries =
+          com.google.cloud.tools.libraries.CloudLibraries.getCloudLibraries();
+      Builder<String> setBuilder = ImmutableSet.builder();
+      
+      for (CloudLibrary library : cloudLibraries) {
+        List<CloudLibraryClient> clients = library.getClients();
+        if (clients != null) {
+          for (CloudLibraryClient client : clients) {
+            CloudLibraryClientMavenCoordinates coordinates = client.getMavenCoordinates();
+            if (coordinates != null) {
+              setBuilder.add(coordinates.getArtifactId()); 
+            }
+          }
+        }
+      }
+      
+      artifacts = setBuilder.build();
+    } catch (IOException ex) {
+      artifacts = new HashSet<>();
+    }
+  }
+  
+  static boolean defines(Element dependencyManager, String groupId, String artifactId) {
+    
+    XPath xpath = factory.newXPath();
+    xpath.setNamespaceContext(new Maven4NamespaceContext());
+
+    try {
+      String bomGroupId = (String) xpath.evaluate(
+          "string(./m:dependencies/m:dependency/m:groupId)",
+          dependencyManager,
+          XPathConstants.STRING);
+      String bomArtifactId = (String) xpath.evaluate(
+          "string(./m:dependencies/m:dependency/m:artifactId)",
+          dependencyManager,
+          XPathConstants.STRING);
```

Comment:
Let's say users include another 3rd-party BOM in addition to ours. I guess this code may not work, since it could return the group ID and artifact ID of the other 3rd-party POM?

Reply:
Good point. Let me think about that.
FYI, right now in prod the code is broken with any dependencyManagement section including ours.